Cannot run doctests or build docs for projects containing 'server' methods.

While looking into https://github.com/leptos-rs/cargo-leptos/issues/66 I've discovered it seems we are unable to run doctests or build docs for any project containing `server` methods.

Here is some example output using the `todo_app_sqlite_axum` example.
At the end of this error output you will see that this is coming from a failing `rustdoc` command.

```bash
    Checking leptos_axum v0.1.3 (/home/phillipb/Repositories/leptos-experiments/leptos/integrations/axum)
 Documenting todo_app_sqlite_axum v0.1.0 (/home/phillipb/Repositories/leptos-experiments/leptos/examples/todo_app_sqlite_axum)
error[E0407]: method `call_fn_client` is not a member of trait `leptos::ServerFn`
  --> src/todo.rs:39:1
   |
39 | #[server(GetTodos, "/api")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `leptos::ServerFn`
   |
   = note: this error originates in the attribute macro `server` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0407]: method `call_fn_client` is not a member of trait `leptos::ServerFn`
  --> src/todo.rs:80:1
   |
80 | #[server(AddTodo, "/api")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `leptos::ServerFn`
   |
   = note: this error originates in the attribute macro `server` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0407]: method `call_fn_client` is not a member of trait `leptos::ServerFn`
  --> src/todo.rs:97:1
   |
97 | #[server(DeleteTodo, "/api")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `leptos::ServerFn`
   |
   = note: this error originates in the attribute macro `server` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Compilation failed, aborting rustdoc

For more information about this error, try `rustc --explain E0407`.
error: could not document `todo_app_sqlite_axum`

Caused by:
  process didn't exit successfully: `rustdoc --edition=2021 --crate-type cdylib --crate-type rlib --crate-name todo_app_sqlite_axum src/lib.rs ... (edited for brevity)
```

Here is the command I'm using to run the doctests.

```bash
cd examples/todo_app_sqlite_axum
LEPTOS_OUTPUT_NAME=todo_app_sqlite_axum LEPTOS_SITE_ROOT=target/site LEPTOS_SITE_PKG_DIR=pkg LEPTOS_SITE_ADDR=127.0.0.1:3000 LEPTOS_RELOAD_PORT=3001 LEPTOS_LIB_DIR=. LEPTOS_BIN_DIR=. cargo test --package=todo_app_sqlite_axum --doc --target-dir=target/server --no-default-features --features=ssr
```

Likewise running cargo doc against the example project also fails.  Again it is the rustdoc command being run by cargo that fails complaining about the server function (as shown above).

```bash
cd examples/todo_app_sqlite_axum
LEPTOS_OUTPUT_NAME=todo_app_sqlite_axum LEPTOS_SITE_ROOT=target/site LEPTOS_SITE_PKG_DIR=pkg LEPTOS_SITE_ADDR=127.0.0.1:3000 LEPTOS_RELOAD_PORT=3001 LEPTOS_LIB_DIR=. LEPTOS_BIN_DIR=. cargo doc --no-deps --target-dir=target/server --no-default-features --features=ssr
```

The nature of the errors suggests `rustdoc` is invoking `rustc` in a way that leads to a failed compilation of the `server` macro.

Just wondering if anyone has any suggestions on how to troubleshoot this further?

Thanks.

--- a/leptos_macro/src/server.rs
+++ b/leptos_macro/src/server.rs
@@ -157,14 +157,14 @@ pub fn server_macro_impl(args: proc_macro::TokenStream, s: TokenStream2) -> Resu
                 #encoding
             }
 
-            #[cfg(any(feature = "ssr", doc))]
+            #[cfg(feature = "ssr")]
             fn call_fn(self, cx: ::leptos::Scope) -> std::pin::Pin<Box<dyn std::future::Future<Output = Result<Self::Output, ::leptos::ServerFnError>>>> {
                 let #struct_name { #(#field_names),* } = self;
                 #cx_assign_statement;
                 Box::pin(async move { #fn_name( #cx_fn_arg #(#field_names_2),*).await })
             }
 
-            #[cfg(any(not(feature = "ssr"), doc))]
+            #[cfg(not(feature = "ssr"))]
             fn call_fn_client(self, cx: ::leptos::Scope) -> std::pin::Pin<Box<dyn std::future::Future<Output = Result<Self::Output, ::leptos::ServerFnError>>>> {
                 let #struct_name { #(#field_names_3),* } = self;
                 Box::pin(async move { #fn_name( #cx_fn_arg #(#field_names_4),*).await })

